sqlutil/percentile.go

```diff
@@ -12,24 +12,26 @@ import (
 // ParsePercentileStr coverts a string of percentile in different form to the percentile as decimal.
 // There are 3 kind of forms, like P99.9 has form 0.999, form 99.9 or form 999. All should be parsed as 0.999
 func ParsePercentileStr(percentileStr string) (float64, error) {
-	percentileStr = strings.TrimSpace(percentileStr)
-	f, err := strconv.ParseFloat(percentileStr, 64)
+	s := strings.TrimLeft(strings.TrimSpace(percentileStr), "pP")
+	f, err := strconv.ParseFloat(s, 64)
 	if err != nil {
-		return 0, fmt.Errorf("percentile value could not be parsed into a number: %s ", percentileStr)
+		return 0, fmt.Errorf("error parsing '%s' ('%s') as float64: %s ", percentileStr, s, err)
 	}
 
 	var percentile float64
 	if f < 1 {
 		// percentile of the form 0.999 (P99.9)
 		percentile = f
-	} else if f >= 1 && f <= 100 {
+	} else if f >= 1 && f < 100 {
 		// percentile of the form 99.9 (P99.9)
 		percentile = f / 100.0
+	} else if f == 100.0 {
+		return 100, nil
```

Comment:
Shouldn't return 1 when f == 100.0?

Reply:
I think P1 should be 0.01 after parsing, and P100 should be 1 in this case. This result is used here: https://github.com/cashapp/blip/blob/main/metrics/percona/mysql_qrt_histogram.go#L53 and it is supposed to be in range of [0, 1.0]

Reply:
Yep, you're right; I just saw that too. Fixing and testing.